### DISCUSSION TITLE
Command to build rm from BSDCoreutils

### DISCUSSION THREAD

**Discussion by andy5995** [theimpossibleastronaut/rmw#372] [General]
Note to self:

`gcc -Wall -I../../compat/include rm.c ../../compat/src/pwcache.c ../../compat/unkown_brother/strlcpy.c ../../compat/src/strmode.c -o rm`

Over the last several years, I tried a few times to compile `rm` from GNUCoreutils (I never succeeded). I was able to do it in just a couple minutes using @DiegoMagdaleno / [BSDCoreUtils](https://github.com/DiegoMagdaleno/BSDCoreUtils).

I'm thinking about using it for the purge feature of rmw. I was using my own code, then recently replaced it with a call to system(rm...), but now I realize that calling the binary for each file purged would be far too insane.

I think I'll wanna check with the maintainers before doing so though, just to make sure there are no licensing issues. I might just add BSDCoreutils as a submodule to keep everything intact, including all the licensing and docs included in that repo.

I'm also considering [busybox](https://busybox.net/) but I haven't played it with that for a while and not sure yet what the process is for compiling `rm`.

**Comment by DiegoMagdaleno** [comment]
There are no licensing issues, as your software appears to be GPL while BSDCoreUtils is BSD, you're allowed to mix software from said licenses without any issues.

I'm fine with it and thank you for using it :).

**Reply by andy5995** [reply]
Great! Thanks for maintaining it. :)

**Comment by andy5995** [comment]
@DiegoMagdaleno I got a few minor warnings when compiling with the default flags in my meson build. Would you like me to try and fix them on your repo?

```
../src/bsd-coreutils/rm.c: In function ‘pass’:
../src/bsd-coreutils/rm.c:347:28: warning: comparison of integer expressions of different signedness: ‘off_t’ {aka ‘long int’} and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  347 |                 wlen = len < bsize ? len : bsize;
      |                            ^
../src/bsd-coreutils/rm.c:347:38: warning: operand of ‘?:’ changes signedness from ‘off_t’ {aka ‘long int’} to ‘size_t’ {aka ‘long unsigned int’} due to unsignedness of other operand [-Wsign-compare]
  347 |                 wlen = len < bsize ? len : bsize;
      |                                      ^~~
../src/bsd-coreutils/rm.c:350:42: warning: comparison of integer expressions of different signedness: ‘ssize_t’ {aka ‘long int’} and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  350 |                 if (write(fd, buf, wlen) != wlen)
```

**Reply by DiegoMagdaleno** [reply]
Sure, that would be okay with me.

Doesn't look like the "unsignedness" of size_t is being truly used in the code, as per my understading, write needs to return the same ammount of bytes written as whatever wlen was calculated to be. Same with the off_t case.
